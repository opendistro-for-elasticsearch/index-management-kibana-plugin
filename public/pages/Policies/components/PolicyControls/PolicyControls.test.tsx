/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent, wait } from "@testing-library/react";
// @ts-ignore
import userEvent from "@testing-library/user-event";
import PolicyControls from "./PolicyControls";

describe("<PolicyControls /> spec", () => {
  it("renders the component", async () => {
    const { container } = render(
      <PolicyControls
        activePage={0}
        pageCount={1}
        search={"testing"}
        onSearchChange={() => {}}
        onPageClick={() => {}}
        onRefresh={async () => {}}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onSearchChange when typing", async () => {
    const onSearchChange = jest.fn();
    const { getByPlaceholderText } = render(
      <PolicyControls
        activePage={0}
        pageCount={1}
        search={""}
        onSearchChange={onSearchChange}
        onPageClick={() => {}}
        onRefresh={async () => {}}
      />
    );

    userEvent.type(getByPlaceholderText("Search"), "four");

    expect(onSearchChange).toHaveBeenCalledTimes(4);
  });

  it("shows/hides pagination", async () => {
    const { queryByTestId, rerender } = render(
      <PolicyControls
        activePage={0}
        pageCount={1}
        search={""}
        onSearchChange={() => {}}
        onPageClick={() => {}}
        onRefresh={async () => {}}
      />
    );

    expect(queryByTestId("policyControlsPagination")).toBeNull();

    rerender(
      <PolicyControls
        activePage={0}
        pageCount={2}
        search={""}
        onSearchChange={() => {}}
        onPageClick={() => {}}
        onRefresh={async () => {}}
      />
    );

    expect(queryByTestId("policyControlsPagination")).not.toBeNull();
  });

  it("calls onPageClick when clicking pagination", async () => {
    const onPageClick = jest.fn();
    const { getByTestId } = render(
      <PolicyControls
        activePage={0}
        pageCount={2}
        search={""}
        onSearchChange={() => {}}
        onPageClick={onPageClick}
        onRefresh={async () => {}}
      />
    );

    fireEvent.click(getByTestId("pagination-button-1"));

    expect(onPageClick).toHaveBeenCalledTimes(1);
  });

  it("calls onRefresh on an interval", async () => {
    const onRefresh = jest.fn();
    const { getByTestId } = render(
      <PolicyControls activePage={0} pageCount={2} search={""} onSearchChange={() => {}} onPageClick={() => {}} onRefresh={onRefresh} />
    );

    fireEvent.click(getByTestId("superDatePickerToggleQuickMenuButton"));

    expect(getByTestId("superDatePickerToggleRefreshButton")).toBeDisabled();

    userEvent.type(getByTestId("superDatePickerRefreshIntervalInput"), "1");

    expect(getByTestId("superDatePickerToggleRefreshButton")).toBeEnabled();

    fireEvent.click(getByTestId("superDatePickerToggleRefreshButton"));

    await wait(() => expect(onRefresh).toHaveBeenCalledTimes(2));
  });
});
