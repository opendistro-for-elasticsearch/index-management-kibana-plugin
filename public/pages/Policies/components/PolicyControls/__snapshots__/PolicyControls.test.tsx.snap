// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PolicyControls /> spec renders the component 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
  style="padding: 0px 5px;"
>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormControlLayout euiFormControlLayout--fullWidth"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <input
          class="euiFieldSearch euiFieldSearch--fullWidth"
          placeholder="Search"
          type="search"
          value="testing"
        />
        <div
          class="euiFormControlLayoutIcons"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            EuiIconMock
          </span>
        </div>
        <div
          class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
        >
          <button
            aria-label="Clear input"
            class="euiFormControlLayoutClearButton"
            type="button"
          >
            EuiIconMock
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
    style="max-width: 250px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiSuperDatePicker__flexWrapper euiSuperDatePicker__flexWrapper--isAutoRefreshOnly"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormControlLayout euiFormControlLayout--group euiSuperDatePicker"
        >
          <div
            class="euiPopover euiPopover--anchorDownLeft"
            id="QuickSelectPopover"
          >
            <div
              class="euiPopover__anchor euiQuickSelectPopover__anchor"
            >
              <button
                aria-label="Date quick select"
                class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--iconRight euiFormControlLayout__prepend"
                data-test-subj="superDatePickerToggleQuickMenuButton"
                type="button"
              >
                <span
                  class="euiButtonEmpty__content"
                >
                  EuiIconMock
                  <span
                    class="euiButtonEmpty__text euiQuickSelectPopover__buttonText"
                  >
                    EuiIconMock
                  </span>
                </span>
              </button>
            </div>
          </div>
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <div
              class="euiDatePickerRange euiDatePickerRange--readOnly euiDatePickerRange--inGroup"
            >
              <span
                class="euiSuperDatePicker__prettyFormat"
              >
                Off
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
